mbq/env/__init__.py

```diff
@@ -7,30 +9,73 @@ class EnvException(Exception):
     pass
 
 
+class Environment(Enum):
+    """
+    A simple Enum to provide consistency in how we detect and
+    represent the environment in which a service is running.
+    """
+
+    PRODUCTION = ('prd', 'Production')
+    DEVELOPMENT = ('dev', 'Development')
+    LOCAL = ('lcl', 'Local')
+
+    def __init__(self, short, long):
+        self.short = short
+        self.long = long
+        self._searchable = {short.lower(), long.lower()}
+
+    @classmethod
+    def from_value(cls, value):
+        """
+        Find an Environment using a single `str`.
+
+        Using a typical Enum, finding a member by value would be as
+        simple as MyEnum('the-value') but since Environment members
+        are stored as tuples and we want to look up using `str`s,
+        `from_value` can used to make that lookup in a case-
+        insensitive way.
+
+        >>> assert Environment.from_value('prd') is Environment.from_value('PRODUCTION')
+        """
+
+        for member in list(cls):
+            if value.lower() in member._searchable:
+                return member
+
+        raise ValueError(f"'{value}' is not a valid {cls.__name__}")
+
+
 class Env:
     def get(self, key, default=NOT_PROVIDED, required=True, coerce=NOT_PROVIDED):
         try:
             val = os.environ[key].strip()
-        except KeyError:
+        except KeyError as e:
             if default is not NOT_PROVIDED:
                 return default
 
             if not required:
                 return None
 
-            raise EnvException('Missing key "{}"'.format(key))
+            raise EnvException('Missing key "{}"'.format(key)) from e
 
         if coerce is not NOT_PROVIDED:
             val = coerce(val)
 
         return val
 
+    def get_environment(self, key, default=NOT_PROVIDED, required=True):
+        try:
+            return self.get(key, default=default, required=required, coerce=Environment.from_value)
+        except ValueError as e:
+            if 'is not a valid Environment' in str(e):
```

Comment:
This is crazy gross, why do we need custom handling here at all?

Reply:
Yeah I went back and forth here. It's mostly to keep the API clean so you can always expect an `EnvException` (see `get_int`). Down to remove it but then let's remove it in `get_int` too.

Reply:
If the intent is to always return an `EnvException` then why do we need to branch on the type of error?

Reply:
Granted, exceptions can be raised from anywhere and we're not wrapping everything in a `try/except` but where we know certain exceptions are likely or even intentional, converting the exception to something expected can make interacting with the library easier.

```python
try:
    env.get_environment('ENV_NAME')
except env.EnvException as e:
    # handle everything related to mbq.env
except ValueError:
    # figure out if this is the ValueError from the Enum or something else?
```

Reply:
Changed it to let the ValueError through. I don't feel strongly about it and the traceback should make it fairly straightforward to figure out.